api/client/swarm/init.go

```diff
@@ -36,7 +37,8 @@ func newInitCommand(dockerCli *client.DockerCli) *cobra.Command {
 	}
 
 	flags := cmd.Flags()
-	flags.Var(&opts.listenAddr, "listen-addr", "Listen address")
+	flags.Var(&opts.listenAddr, flagListenAddr, "Listen address")
+	flags.StringVar(&opts.listenInterface, flagListenInterface, "", "Listen on an interface's address")
```

Comment:
I really dont like using `/` as it is just making up a syntax. Thinking about it actually `:` is fine though, a it is only valid in the archaic alias syntax, and you should not be getting those pseudo interface names from the Go APIs I don't think. But the whole idea of specifying an interface is pretty horrible I think, seems like a case where the user should just make a script to choose the right addresses, there are too many edge cases.